adapters/restclient/lib/opentelemetry/adapters/restclient/patches/request.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module RestClient
+      module Patches
+        # Module to prepend to RestClient::Request for instrumentation
+        module Request
+          def execute(&block)
+            return super(&block) if disable_span_reporting?
+
+            trace_request do |span|
+              propagator.inject(span.context, processed_headers)
```

Comment:
Is there any problem moving this to line 41? It would make things easier to reason through if so.